code/drasil-printers/Language/Drasil/TeX/Helpers.hs

```diff
@@ -181,33 +180,31 @@ sec :: Int -> D -> D
 sec d b1 = genSec d <> br b1
 
 subscript, superscript :: D -> D -> D
-subscript a b = a <> pure H.unders <> br b
-superscript a b = a <> pure H.hat <> br b
+subscript   a b = a <> pure H.unders <> br b
+superscript a b = a <> pure H.hat    <> br b
 
 -- grave, acute :: Char -> D
 -- grave c = (pure $ text "\\`{") <> pure (TP.char c) <> (pure $ text "}")
 -- acute c = (pure $ text "\\'{") <> pure (TP.char c) <> (pure $ text "}")
 
 -- Macro / Command def'n --
 --TeX--
-srsComms, lpmComms, bullet, counter, ddefnum, ddref, colAw, colBw, arrayS
- , modcounter, modnum :: D
-srsComms = bullet %% counter %% ddefnum %% ddref %% colAw %% colBw %% arrayS
-lpmComms = pure $ text ""
+bullet, counter, ddefnum, ddref, colAw, colBw, arrayS, modcounter, modnum, newline :: D
 
-counter       = count "datadefnum"
-modcounter    = count "modnum"
+counter    = count "datadefnum"
+modcounter = count "modnum"
 
 bullet  = comm "blt"             "- "                Nothing
 ddefnum = comm "ddthedatadefnum" "MG\\thedatadefnum" Nothing
 ddref   = comm "ddref"           "MG\\ref{#1}"       (Just "1")
 colAw   = comm "colAwidth"       "0.2\\textwidth"    Nothing
 colBw   = comm "colBwidth"       "0.73\\textwidth"   Nothing
-
-modnum    = comm "mthemodnum"        "M\\themodnum"        Nothing
+modnum  = comm "mthemodnum"      "M\\themodnum"      Nothing
 
 arrayS  = renewcomm "arraystretch" "1.2"
 
+newline = pure $ text ""
```

Comment:
So this seems wrong now. I'm guessing that this doesn't act as a newline anymore!  We likely just want to remove calls to this fake `newline`.

Reply:
This is actually what makes this PR work. Before, I added a `Paragraph (S "")` in between consecutive `Paragraph`s, which generated an empty line in the LaTeX, leading to a new line in the generated PDF. This is essentially doing the same thing. As per your other comment, yes, in essence the `%%` is adding a newline, because it concatenates two `D`s with a newline character, and the second `D` is an empty line. As mentioned before, I tried using `empty`, but that is the identity element, so it doesn't actually do anything. Is there an alternative that would be less ambiguous?

Reply:
I think there is a `newline` type of command in the HughesPJ package, that would be the right way to do this.

Reply:
There isn't a `newline`-esque command in the `Text.PrettyPrint` package. The closest thing is `($+$)`, which adds two `Doc`s together on separate lines. This gives us one new line, but we need another empty line to actually have the PDF have a new line.

Related question; why do we have both of these functions? I've tested it out, and it seems that they are completely interchangeable:
```haskell
(%%) :: D -> D -> D
(%%) = mappend

($+$) :: D -> D -> D
($+$) = liftA2 (TP.$+$)
```

Reply:
First part: Ok.  But we still need to do things differently, because 
```haskell
pure $ text ""
```
isn't a "newline", it just happens to work after a `(%%)`.  If the type was more like
```haskell
-- add newline
newline :: D -> D
newline s = s %% (pure $ text "")
```
then that couldn't be mis-used.

Second part: no idea! If `mappend` in `Text.PrettyPrint` is defined as `$+$` (or the other way around), then indeed they are equivalent, and one should be removed.  But I need to know with more certainty first.

Reply:
I revised `newline` in 3dd9220388b5e45a204b81282f37ea9c7713ea0a.

Do you think the `(%%)` vs. `($+$)` discussion is worth having an issue for and discussing it there?

Reply:
I looked into it, and found the key difference between the two functions. Issue created at #1758.